### PR TITLE
Fix client hangs on BF.INSERT with multiple new items when non scaling filter is full

### LINKED ISSUE #216: BF.INSERT with multiple new items when non scaling filter is full leads to clients hanging indefinetly for array elements that will not be sent
You can easily replicate this with Docker and redis-cli:
Start RedisBloom container:
```
docker run -p 6379:6379 redislabs/rebloom:latest
```
on redis-cli:
```
redis-cli BF.INSERT nonscaling_err CAPACITY 3 NONSCALING ITEMS a b c d e
```

tcp stream request portion
```
*11
$9
BF.INSERT
$14
nonscaling_err
$8
CAPACITY
$1
3
$10
NONSCALING
$5
ITEMS
$1
a
$1
b
$1
c
$1
d
$1
e
```
tcp stream reply ( notice that we're explicitly stating we'll reply 5 array elements and we only reply 4 ( up to the first error ) 
```
*5
:1
:1
:1
-Non scaling filter is full
```

This issue should be fixed at `bfInsertCommon` for which I will send PR right away

### PR DESCRIPTION
**Pull request by filipecosta90**
fixes #216

### CODE REVIEW

File: src/rebloom.c [hunk 1]
```diff
@@ -229,20 +230,28 @@ static int bfInsertCommon(RedisModuleCtx *ctx, RedisModuleString *keystr, RedisM
...
-    for (size_t ii = 0; ii < nitems; ++ii) {
+    size_t array_len = 0;
+    for (size_t ii = 0; ii < nitems && result == REDISMODULE_OK; ++ii) {
         size_t n;
         const char *s = RedisModule_StringPtrLen(items[ii], &n);
         int rv = SBChain_Add(sb, s, n);
         if (rv == -2) { // decide if to make into an error
-            return RedisModule_ReplyWithError(ctx, "Non scaling filter is full");
+            RedisModule_ReplyWithError(ctx, "Non scaling filter is full");
+            result = REDISMODULE_ERR;
```
Comment (ashtul) [inline src/rebloom.c:243]: what are you doing with the error reply?
I think we should still return OK and the user will see the error in the array.

File: tests/pytests.py [hunk 2]
```diff
@@ -297,6 +297,13 @@ def test_no_scaling(self):
         self.assertOk(self.cmd('bf.reserve bfscale 0.001 1000'))
         self.assertLess(self.cmd('bf.info bfnonscale')[3], self.cmd('bf.info bfscale')[3])
 
+    def test_nonscaling_err(self):
+        capacity = 3
+        self.assertEqual(self.cmd('BF.INSERT nonscaling_err CAPACITY 3 NONSCALING ITEMS a b c'), [1L,1L,1L])
+        # clients hangs
+        self.cmd('BF.INSERT nonscaling_err ITEMS a b c d d')
```
Comment (ashtul) [inline tests/pytests.py:304]: please test for reply string

File: src/rebloom.c (hunk 1 above)
Comment (filipecosta90) [reply src/rebloom.c:243]: @ashtul to keep the same behaviour as before, we need to return `REDISMODULE_ERR`

File: tests/pytests.py (hunk 2 above)
Comment (filipecosta90) [reply tests/pytests.py]: agree. fixed in the new commit

File: src/rebloom.c (hunk 1 above)
Comment (ashtul) [reply src/rebloom.c:243]: `RedisModule_ReplyWithError` always returns `REDISMODULE_OK`

Comment (filipecosta90) [reply src/rebloom.c]: you're totally right. fixed on the new commit

**Pull request merged**